qiskit_experiments/calibration/calibrations.py

```diff
@@ -0,0 +1,438 @@
+
+        param = self._parameter_map[(sched_name, param_name)]
+
+        if qubits not in self._params[param]:
+            raise CalibrationError(f"No parameter value for {param} and qubits {qubits}.")
+
+        if valid_only:
+            candidates = [p for p in self._params[param][qubits] if p.valid]
+        else:
+            candidates = self._params[param][qubits]
+
+        candidates = [candidate for candidate in candidates if candidate.group == group]
+
+        if cutoff_date:
+            candidates = [val for val in candidates if val.date_time <= cutoff_date]
+
+        if len(candidates) == 0:
+            msg = (
+                f"No candidate parameter values for {param_name} in calibration group "
+                f"{group} on qubits {qubits} in schedule {sched_name} "
+            )
+
+            if cutoff_date:
+                msg += f" Cutoff date: {cutoff_date}"
+
+            raise CalibrationError(msg)
+
+        candidates.sort(key=lambda x: x.date_time)
+
+        return candidates[-1].value
+
+    def get_schedule(
+        self,
+        name: str,
+        qubits: Tuple[int, ...],
+        free_params: List[str] = None,
+        group: Optional[str] = "default",
+    ) -> Schedule:
+        """
+        Args:
+            name: The name of the schedule to get.
+            qubits: The qubits for which to get the schedule.
+            free_params: The parameters that should remain unassigned.
+            group: The calibration group from which to draw the
+                parameters. If not specifies this defaults to the 'default' group.
+
+        Returns:
+            schedule: A deep copy of the template schedule with all parameters assigned
+                except for those specified by free_params.
+
+        Raises:
+            CalibrationError: if the name of the schedule is not known.
+        """
+
+        # Get the schedule and deepcopy it to prevent binding from removing
+        # the parametric nature of the schedule.
+        if name not in self._schedules:
+            raise CalibrationError("Schedule %s is not defined." % name)
+
+        sched = copy.deepcopy(self._schedules[name])
+
+        # Retrieve the channel indices based on the qubits and bind them.
+        binding_dict = {}
+        for ch in sched.channels:
+            if ch.is_parameterized():
+                binding_dict[ch.index] = self._get_channel_index(qubits, ch)
+
+        sched.assign_parameters(binding_dict)
+
+        # Loop through the remaining parameters in the schedule, get their values and bind.
+        if free_params is None:
+            free_params = []
+
+        binding_dict = {}
+        for param in sched.parameters:
+            if param.name not in free_params:
+                binding_dict[param] = self.get_parameter_value(
+                    param.name, qubits, name, group=group
+                )
+
+        sched.assign_parameters(binding_dict)
+
+        return sched
+
+    def get_circuit(
```

Comment:
I think schedule should be defined for _gate_, not circuit. It looks like you want to use `add_calibration` to create a self-contained instruction as a program, but this means we cannot override existing circuit instruction. For example, when we apply calibration to QV experiment, we need to prepare another QV experiment generator to construct experiment circuit with `circuit.compose`. We need dedicated transpile passes since a circuit doesn't have unitary definition. I don't think this is right direction. What we need in this class is a method to return parameter-assigned schedule by taking a `Gate` and qubit as inputs. `add_calibration` should be called from the actual experiment circuit generator.

Reply:
I see your point and I would solve this differently. `get_circuit` is used here for the calibration experiments. Once you are done with the calibration I would add a method to export the calibrations into a backend, e.g. in which the `instruction_schedule_map` is replaced. Then run the QV in the normal flow. E.g. something like
```python
def export_backend(self) -> 'backend':
    """Export the calibrations into a useable backend."""
```

Reply:
I still think this method should be utility function in the calibration experiments, because we are calibrating instructions (=`Gate`), not programs (=`QuantumCircuit`).

```python
def get_circuit(name, qubits, group, calibration) -> QuantumCircuit:
   ...
```

I really like the idea of `export_backend`. We need to check if the backend compiler refers to the attached instmap (I don't think it does).

Reply:
It could be. I don't want to through it out quite yet so I made it private for now. See here: 9de4086